cartridge/invalid-format.lua

```diff
@@ -0,0 +1,68 @@
+local log = require('log')
+
+local illegal_types = {
+    [''] = true,
+    ['n'] = true,
+    ['nu'] = true,
+    ['s'] = true,
+    ['st'] = true,
+}
+
+rawset(_G, '__cartridge_invalid_format_spaces', {})
+
+-- _space trigger.
+local function before_replace(_, tuple)
+    if tuple == nil then return tuple end
+
+    local name = tuple[3]
+    local format = tuple[7]
+
+    for _, field_def in ipairs(format) do
+        if illegal_types[field_def.type] then
+            _G.__cartridge_invalid_format_spaces[name] = true
+        end
```

Comment:
I would also look at `field_def[2]`. AFAIU a user may have a space created, when the old format representation was the default one.